### PR TITLE
added configuration 'http_proxy' to allow the usage of a proxy

### PR DESCRIPTION
**Pull request by don41382**

### CODE REVIEW

**Comment by mbohlool** [comment]
LGTM. is it possible to add a test for this?

**Comment by mbohlool** [comment]
@don41382 thanks for the PR. If you don't have time to figure out the test, let me know and I can look into adding a test if possible.

**Comment by don41382** [comment]
@mbohlool sorry for the delayed response. I actually not sure how to test this. I am not a big python programmer. If you could show me how I would appreciate this.

**Comment by mbohlool** [comment]
You can use some mock libraries but that I am not sure what is the best practice in python too. We can merge this without test. Thanks.

File: kubernetes/client/configuration.py [hunk 1]
```diff
@@ -89,6 +89,9 @@ def __init__(self):
         # Set this to True/False to enable/disable SSL hostname verification.
         self.assert_hostname = None
 
+	# http proxy setting
+        self.http_proxy = None
```
Comment (mbohlool) [inline kubernetes/client/configuration.py]: minor: Can you rename this to `http_proxy_url` to self-document the type?

**Comment by mbohlool** [comment]
I just added a minor comment. Will merge after you address it. please stash new changes in the existing commit. Thanks.

File: kubernetes/client/configuration.py (hunk 1 above)
Comment (don41382) [inline kubernetes/client/configuration.py]: 👍

**Comment by mbohlool** [comment]
LGTM. Please rebase upstream/master and stash two commits together (use git rebase -i HEAD~2).

**Comment by mbohlool** [comment]
@don41382 Sorry. things are moving around in the client because of the restructuring of the client. You need to send this change to https://github.com/kubernetes-client/python-base now.

**Comment by mbohlool** [comment]
as it was my fault moving things around while you were waiting for this to merge, this is an script to do the pull request automatically, just set your github username and make sure you have `hub` tool installed:

```bash
# make sure you have hub tool installed from here https://hub.github.com/
# run this in a temp folder

export GITHUB_USER=YOUR_GITHUB_USER

git clone https://github.com/kubernetes-client/python-base
cd python-base
hub fork
curl -o patch.diff https://patch-diff.githubusercontent.com/raw/kubernetes-incubator/client-python/pull/206.patch
sed -i'' "s.kubernetes/client/..g" patch.diff
git apply patch.diff
git commit -m "added configuration 'http_proxy' to allow the usage of a proxy"
git push $GITHUB_USER master
hub pull-request
```

**Comment by mbohlool** [comment]
closing in favor of https://github.com/kubernetes-client/python-base/pull/8

**Pull request closed without merging**